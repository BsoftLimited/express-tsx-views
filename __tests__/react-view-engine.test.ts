import { resolve } from 'path'
import request from 'supertest'
import { setupReactViews } from '..'
import { app } from '../example/app'

describe('react-view-engine', () => {
  test('Rendering', async () => {
    const response = await request(app).get('/').expect(200)

    expect(response.text).toMatchInlineSnapshot(`
      "<!DOCTYPE html>
      <html lang=\\"de\\">
        <head>
          <style>
            h1{color:red}
          </style>
          <meta charset=\\"UTF-8\\" />
          <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
          <title>Test</title>
        </head>
        <body>
          <h1>Test</h1>
          <p>Some component:</p>
          Hello from MyComponent! Provided prop: foo
        </body>
      </html>
      "
    `)
  })

  it('Rendering with locals', async () => {
    const response = await request(app).get('/with-locals').expect(200)

    expect(response.text).toMatchInlineSnapshot(`
      "<!DOCTYPE html>
      <html>
        <head>
          <style>
            h1{color:red}
          </style>
          <meta charset=\\"UTF-8\\" />
          <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
          <title>from locals</title>
        </head>
        <body>
          <h1>from locals</h1>
          <p>Some component:</p>
          Hello from MyComponent! Provided prop: foo
        </body>
      </html>
      "
    `)
  })

  it('without DOCTYPE', async () => {
    setupReactViews(app, {
      viewsDirectory: resolve(__dirname, '../example/views'),
      prettify: true,
      doctype: '',
    })

    const response = await request(app).get('/with-locals').expect(200)
    expect(response.text).toMatchInlineSnapshot(`
      "<html>
        <head>
          <style>
            <!-- CSS -->
          </style>
          <meta charset=\\"UTF-8\\" />
          <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
          <title>from locals</title>
        </head>
        <body>
          <h1>from locals</h1>
          <p>Some component:</p>
          Hello from MyComponent! Provided prop: foo
        </body>
      </html>
      "
    `)
  })
})
