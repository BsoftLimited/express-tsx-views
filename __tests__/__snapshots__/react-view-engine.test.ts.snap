// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-view-engine Rendering 1`] = `
"<!DOCTYPE html>
<html lang=\\"de\\">
  <head>
    <style>
      h1{color:red}
    </style>
    <meta charset=\\"UTF-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>Test</title>
  </head>
  <body>
    <h1>Test</h1>
    <p>Some component:</p>
    Hello from MyComponent! Provided prop:
    <figure><pre>&quot;foo&quot;</pre></figure>
    <hr />
    <figure>
      <figcaption>props:</figcaption>
      <pre>
{
  &quot;title&quot;: &quot;Test&quot;,
  &quot;lang&quot;: &quot;de&quot;
}</pre
      >
    </figure>
    <hr />
    <figure>
      <figcaption>context1:</figcaption>
      <pre>
{
  &quot;id&quot;: 1,
  &quot;name&quot;: &quot;test&quot;
}</pre
      >
    </figure>
    <hr />
    <figure>
      <figcaption>context2:</figcaption>
      <pre>
{
  &quot;someProperty&quot;: &quot;mausi&quot;
}</pre
      >
    </figure>
  </body>
</html>
"
`;

exports[`react-view-engine Rendering with locals 1`] = `
"<!DOCTYPE html>
<html>
  <head>
    <style>
      h1{color:red}
    </style>
    <meta charset=\\"UTF-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <title>from locals</title>
  </head>
  <body>
    <h1>from locals</h1>
    <p>Some component:</p>
    Hello from MyComponent! Provided prop:
    <figure><pre>&quot;foo&quot;</pre></figure>
    <hr />
    <figure>
      <figcaption>props:</figcaption>
      <pre>
{
  &quot;title&quot;: &quot;from locals&quot;
}</pre
      >
    </figure>
    <hr />
    <figure>
      <figcaption>context1:</figcaption>
      <pre>
{
  &quot;id&quot;: 1,
  &quot;name&quot;: &quot;test&quot;
}</pre
      >
    </figure>
    <hr />
    <figure>
      <figcaption>context2:</figcaption>
      <pre>
{
  &quot;someProperty&quot;: &quot;mausi&quot;
}</pre
      >
    </figure>
  </body>
</html>
"
`;
